lib/debug/server_dap.rb

```diff
@@ -274,188 +274,233 @@ def recv_request
       retry
     end
 
-    def process
-      while req = recv_request
-        raise "not a request: #{req.inspect}" unless req['type'] == 'request'
-        args = req.dig('arguments')
-
-        case req['command']
+    class RequestHandler
```

Comment:
(using a class to hold only a single field may be unnecessary)

Reply:
How do you envision us adding metrics through this class?

Reply:
> How do you envision us adding metrics through this class?

One approach could be to add a command line option, e,g. `--request-handler="Shopify::RequestHandler"`, allowing a custom class to provided. 

Maybe @st0012 already has something else in mind though.

Reply:
We may not even need this class at all. We can override register_request
```ruby
def self.register_request *names, &b
      instrumented = -> { 
        measure_stuff do
          b.call
        end
      }
      names.each { |name| REGISTERED_REQUESTS[name] = instrumented }
end
```

Reply:
Yeah that would also work, but I worry it may not be resilient to future changes within `debug`.